gui/src/main/tunnel-state.ts

```diff
@@ -20,10 +20,15 @@ export default class TunnelStateHandler {
   // Scheduler for discarding the assumed next state.
   private tunnelStateFallbackScheduler = new Scheduler();
 
+  private receivedFullDiskError = false;
+
```

Comment:
Alternatively, we could just call this `shutdownDaemonWhenGuiCloses`

Reply:
IMO, that would be worse. As is, `TunnelStateHandler` doesn't need to care what the boolean is used for. If it were renamed to `shutdownDaemonWhenGuiCloses` then it would couple it (slightly) more with the main class.